### PR TITLE
Prevent Writer from usage outside of the change block

### LINKED ISSUE #1212: Check if writer is not used outside a change block
The model writer should be used only inside `enqueueChange`, `change` or post fixer block. It can be passed to the inner function, but should not be stored in a variable and used outside of the proper block. Fortunately, we should be able to check it easily. 

Each writer method, when called, should check if `this.model._currentWritter == this` and throw exception if it is not true.

### PR DESCRIPTION
**Pull request by ma2ciek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Prevented `Writer` from usage outside of the `change` block. Closes #1212.

### CODE REVIEW

**Comment by ma2ciek** [comment]
Few tests for errors miss. I'm on it.

File: src/model/writer.js [hunk 1]
```diff
@@ -150,6 +150,10 @@ export default class Writer {
 	 * second parameter is a {@link module:engine/model/item~Item model item}.
 	 */
 	insert( item, itemOrPosition, offset ) {
+		if ( this.model._currentWriter !== this ) {
+			throw new CKEditorError( 'writer-detached-writer-tries-to-modify-model: Detached writer tries to modify the model' );
+		}
```
Comment (pjasiun) [inline src/model/writer.js]: Wouldn't it be better to have it in a private method?

Comment (ma2ciek) [reply src/model/writer.js]: Probably yes. I moved it to the separate private method.

**Comment by ma2ciek** [comment]
Ready for the review.

**Pull request merged**